File: backend/src/main/scala/review/api/SubmitEligibilityReviewApi.scala
package review.api

import domain.shared.given

import cats.effect.IO
import domain.review.EligibilityReviewRequest
import domain.shared.{DeliveryAppState, EligibilityReviewTarget, RiderId, StoreId, UserRole}
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import review.app.submitEligibilityReview
import shared.api.routing.*
import shared.app.*

val submitEligibilityReviewApi: FixedMethodApi0[DeliveryAppState] =
  jsonPostApi0[EligibilityReviewRequest, DeliveryAppState](
    routeSegment("api"),
    routeSegment("delivery"),
    routeSegment("eligibility-reviews"),
  )

File: frontend/src/apis/review/SubmitEligibilityReviewApi.ts
import type { DeliveryAppState, EligibilityReviewRequest } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryState } from '@/system/api/DeliveryStateClient'
import { defineJsonPostApi0, routeSegment } from '@/system/api/TypedApiDefinitions'

export const submitEligibilityReviewApiDefinition = defineJsonPostApi0<EligibilityReviewRequest, DeliveryAppState>([
    routeSegment('api'),
    routeSegment('delivery'),
    routeSegment('eligibility-reviews'),
  ])

export function submitEligibilityReview(payload: EligibilityReviewRequest) {
  return postNormalizedDeliveryState(submitEligibilityReviewApiDefinition, payload)
}
